### ISSUE TITLE
SubtleCrypto methods return PromiseLike instead of Promise

### ISSUE THREAD

**Issue by taymonbeal** [microsoft/TypeScript#37756]
**TypeScript Version:**  3.9.0-dev.20200328

**Search Terms:**
WebCrypto, web crypto, SubtleCrypto, subtle, PromiseLike

**Expected behavior:**
Should compile with no errors.

**Actual behavior:**
```
Type 'PromiseLike<CryptoKey>' is missing the following properties from type 'Promise<CryptoKey>': catch, [Symbol.toStringTag]
```

**Related Issues:**
microsoft/TSJS-lib-generator#843 (pull request that should fix this)
microsoft/TSJS-lib-generator#60
microsoft/TSJS-lib-generator#47

**Code**
```ts
const keyPromise: Promise<CryptoKey> = crypto.subtle.generateKey({name: 'AES-GCM', length: 256}, true, ['encrypt']);
```

<details><summary><b>Output</b></summary>

```ts
"use strict";
const keyPromise = crypto.subtle.generateKey({ name: 'AES-GCM', length: 256 }, true, ['encrypt']);

```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "alwaysStrict": true,
    "esModuleInterop": true,
    "declaration": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "moduleResolution": 2,
    "target": "ES2017",
    "jsx": "React",
    "module": "ESNext"
  }
}
```

</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/v2/en/play?ts=3.9.0-dev.20200328#code/MYewdgzgLgBA1gUwJ4AUBOIC2BLCCBcM6WuCAPAMJpIAOUIA0sgHwwC8Mw1dIAdBAFcARlAA2CXgHMEYBGgCGUBEyQAKAN5h5mAjADkAQQCiAZQC0AcQoBZPQBoY4sJKgALQgCYArADYAvg5QaAIIDgDaejJctFB6ALoAlADcQA)

**Comment by taymonbeal** [comment]
I've already sent a PR to fix this at microsoft/TSJS-lib-generator#843. It just needs to be reviewed.

**Comment by jakebailey** [comment]
This was fixed in #39052.
